### PR TITLE
Add rename-command config

### PR DESCRIPTION
**Pull request by karelrooted**

### CODE REVIEW

File: src/config.cc [hunk 1]
```diff
@@ -266,6 +266,27 @@ void Config::initFieldCallback() {
...
         return Status::OK();
       }},
+      {"rename-command", [this](Server *srv, const std::string &k, const std::string &v) -> Status {
+        if (v.empty()) {
+          return Status::OK();
+        }
+        std::vector<std::string> args;
+        Util::Split(v, " \t", &args);
+        if (args.size() != 2 || args[0] == args[1]) {
+          return Status(Status::NotOK, "invalid rename-command format");
```
Comment (ShooterIT) [inline src/config.cc]:
```suggestion
          return Status(Status::NotOK, "Invalid rename-command format");
```

File: src/config.h [hunk 2]
```diff
@@ -83,6 +83,8 @@ struct Config{
   Cron bgsave_cron;
   CompactionCheckerRange compaction_checker_range{-1, -1};
   std::map<std::string, std::string> tokens;
+  std::string rename_command_;
+  std::string rename_command_new_name_;
```
Comment (git-hulk) [inline src/config.h]: Should we allow renaming more than one command?

Comment (karelrooted) [reply src/config.h]: redis only allow rename one command ,  kvrocks allow rename multi command ?

Comment (git-hulk) [reply src/config.h]: Redis allows renaming multi commands indeed

**Review by git-hulk — DISMISSED** [review]

File: src/config.cc [hunk 3]
```diff
@@ -175,6 +176,27 @@ void Config::initFieldValidator() {
         compaction_checker_range.Stop = stop;
         return Status::OK();
       }},
+      {"rename-command", [](const std::string &k, const std::string &v) -> Status {
+        std::vector<std::string> args;
+        Util::Split(v, " \t", &args);
+        if (args.size() != 2 || args[0] == args[1]) {
```
Comment (ShooterIT) [inline src/config.cc]: I don't think it is an error if renamed name is the same as origin name

File: kvrocks.conf [hunk 4]
```diff
@@ -270,6 +270,22 @@ compaction-checker-range 0-7
 # e.g. bgsave-cron 0 3 * * * 0 4 * * *
 # would bgsave the db at 3am and 4am everyday
 
+# Command renaming.
+#
+# It is possible to change the name of dangerous commands in a shared
+# environment. For instance the Keys command may be renamed into something
```
Comment (ShooterIT) [inline kvrocks.conf]:
```suggestion
# environment. For instance the KEYS command may be renamed into something
```
just the same as the following

File: src/config.cc [hunk 5]
```diff
@@ -240,7 +260,8 @@ void Config::initFieldCallback() {
           if (cmd == "*") {
             profiling_sample_all_commands = true;
             profiling_sample_commands.clear();
-          } else if (srv->IsCommandExists(cmd)) {
+          } else if (Redis::IsCommandExists(cmd)) {
```
Comment (ShooterIT) [inline src/config.cc:263]: So we must ask users to use origin name, kvrocks won't profile renamed command?

**Comment by ShooterIT** [comment]
@karelrooted And could you supplement some tcl tests

File: src/config.cc (hunk 5 above)
Comment (karelrooted) [reply src/config.cc:263]: redis_request.cc use isProfilingEnabled(original_cmd_name) (line 205), so profile_sample_commands need to use original cmd name

**Review by ShooterIT — DISMISSED** [review]

**Review by git-hulk — DISMISSED** [review]

**Comment by karelrooted** [comment]
> @karelrooted And could you supplement some tcl tests

add a rename one command testcase,  the current code base will not allow a multi rename-command directive,  so rename-two-command  testcase is not included

**Review by git-hulk — APPROVED** [review]

**Review by ShooterIT — APPROVED** [review]

**Pull request merged**